docs/cluster-configuration.md

```diff
@@ -0,0 +1,233 @@
+
+This example would add an entry to `/etc/hosts` on every node to override the IP address of `hostname.sample.mit.edu` to
+be the address `18.0.0.1`. This used to be more useful, but now entries are automatically added to this for the nodes
+within the cluster, so it is more useless.
+
+## Root administrator configuration
+
+Sample section:
+
+    root-admins:
+      - example/root@ATHENA.MIT.EDU
+
+The root administrators in a cluster are the administrators who are granted complete access -- that is, they can
+directly SSH into any node and do whatever they want. This is a list of the Kerberos principals of the root admins.
+
+If this section is empty, it means that Kerberos administrative authentication will be disabled.
+
+Note that, in all cases, root admins can also establish access by simply having the disaster recovery key. This is the
+primary way to establish access when Kerberos authentication is not involved.
+
+## Node configuration
+
+Sample section:
+
+    nodes:
+      - hostname: master-hostname
+        ip: 18.0.0.2
+        kind: master
+
+      - hostname: worker-hostname
+        ip: 18.0.0.3
+        kind: worker
+
+      - hostname: supervisor-hostname
+        ip: 18.0.0.4
+        kind: supervisor
+
+This is a list of the nodes that are part of the cluster. When nodes are configured and installed, this mapping will be
+used to determine what mode they should be placed in, and the supervisor and master node IP addresses will be
+substituted into various configuration files.
+
+In theory, you should be able to update this section to change, add, or remove nodes, but this is not fully supported at
+this time.
+
+Each entry has the following fields:
+
+ * `hostname`: the hostname (under the external domain specified previously) that the node should be accessible at. This
+   does not actually need to be configured in DNS, because it will be coded into /etc/hosts on each node for the time
+   being.
+
+ * `ip`: the statically-assigned IP address for this node. This address must be the same address that will be entered
+   when running the installer for the node.
+
+ * `kind`: the type of node that this should be.
+
+The kinds of nodes are:
+
+ * `supervisor`: a node that is not part of the Kubernetes cluster proper, but assists with its setup and
+   reconfiguration.
+
+   Only a single supervisor node is supported at this time.
+
+   A supervisor node does not need to be up for the regular operations of the cluster, but will need to be up at least
+   intermittently to allow key renewal to occur on the other nodes.
+
+ * `master`: a node that runs the management services of the Kubernetes cluster. These nodes do not run user-provided
+   code, but do run a full Kubernetes node stack, in addition to the management services.
+
+ * `worker`: a node that runs user containers for the Kubernetes cluster.
+
+The number of master nodes should generally be one more than a power of two, because they will need quorum for their
```

Comment:
Shouldn't this just be an odd number rather than 1 + power of 2?  What am I missing here?

Reply:
Yeah, I should really have just said that. I don't know why I didn't.